docs/markdown/DevGuide_Data_Memory.md

```diff
@@ -0,0 +1,85 @@
+# Data Structures
+In Inviwo the main core data structures (`Volume`, `Image`, `Layer`, `Mesh`, `Buffer`) use a pattern of handles and representations. The volume data structure for example has a handle class called `Volume`. The handle class by itself does not have any data, it only stores metadata, and a list of representations. The representations is where the actual data is stored. The following first explains the basic data structures and how they relate, then shows what different representations exist and how to use them.
+
+## The different structures
+In Inviwo, the `Volume` class wraps volumetric data, more specifically data in a structured grid. This corresponds to what you would expect from a rank 3 tensor (or rank 4 if you have a multi-channel volume).
+
+The `Image` class is used to wrap *Frame buffers*, which are in essence packages for multiple images with potentially different shapes and data types. The actual single images inside those frame buffers are represented by a `Layer` in Inviwo.
+As an effect, most `Image` ports in Inviwo actually transfer multiple images, or one frame buffer, usually consisting of a color layer, depth layer and optionally a picking layer. In this example the `Image` would contain a `UInt8` color layer, a `Float32` depth layer and a `UInt8` picking layer. The picking layer basically encodes object instance IDs in color, so that a lookup in the picking layer gives the object ID for the pixel of interest. This is used for example to drag'n'drop objects in 3D space.
+Of course you can add more layers to the images in your custom processors as necessary.
+
+The third big data structure in Inviwo is the `Mesh` and its `Buffer`s. A `Buffer` is basically a wrapper for some linearly stored data and the `Mesh` holds a list of index buffers and other buffers, where you can add your data accordingly. That means when you load in a standard mesh, the resulting `Mesh` has the appropriate set of index buffers, a vertex buffer and probably a normal buffer.
```

Comment:
I would say Index buffers, and a set of data buffers, vertices, normals, colors, texture coordinates, and so on.